src/main/java/at/hannibal2/skyhanni/data/PetAPI.kt

```diff
@@ -1,13 +1,23 @@
 package at.hannibal2.skyhanni.data
 
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
 import at.hannibal2.skyhanni.utils.StringUtils.matches
 import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
 
 object PetAPI {
-    private val petMenuPattern by RepoPattern.pattern(
-        "misc.pet.menu.title",
+    private val patternGroup = RepoPattern.group("misc.pet")
+    private val petMenuPattern by patternGroup.pattern(
+        "menu.title",
         "Pets(?: \\(\\d+/\\d+\\) )?"
     )
+    private val petItemName by patternGroup.pattern(
+        "item.name",
+        "§.\\[Lvl (?<level>\\d)] (?<name>.*)"
+    )
+    private val neuRepoPetItemName by patternGroup.pattern(
+        "item.name.neu.format",
+        "(§f§f)?§7\\[Lvl 1➡(100|200)] (?<name>.*)"
+    )
```

Comment:
Do we really need two patterns?
Since it could simply be done with this pattern:
```
"(?:§.)*\\[Lvl (?:1➡)?(?<level>\\d)] (?<name>.*)"
```

Reply:
I split it into two patterns, as we don't know the level from the second one, and having a nullable group felt wrong to me. But thinking more about it, it's fine, and even cleaner for the future. I plan on merging all pet related patterns on SkyHanni (search for `petLevelPattern`, there are 2 other classes using the same regex) into the PetAPI class.

Reply:
your one regex solution breaks here: https://regex101.com/r/4uYqMg/1

Reply:
I think keeping it as two and then eventually moving it all to petAPI is fine